lib/msal-browser/src/broker/client/BrokerClientApplication.ts

```diff
@@ -188,7 +210,11 @@ export class BrokerClientApplication extends ClientApplication {
      */
     private async brokeredPopupRequest(validMessage: BrokerAuthRequest, clientPort: MessagePort): Promise<void> {
         try {
-            const response: BrokerAuthenticationResult = (await this.acquireTokenPopup(validMessage.request as PopupRequest)) as BrokerAuthenticationResult;
+            const popupRequest = validMessage.request as BrokerPopupRequest;
+            popupRequest.redirectUri = validMessage.embeddedAppRedirectUri;
+            popupRequest.embeddedAppClientId = validMessage.embeddedClientId;
+            popupRequest.brokerRedirectUri = this.getRedirectUri();
+            const response: BrokerAuthenticationResult = (await this.acquireTokenPopup(popupRequest)) as BrokerAuthenticationResult;
```

Comment:
Don't think you need this since you're casting the value

```suggestion
            const response = (await this.acquireTokenPopup(popupRequest)) as BrokerAuthenticationResult;
```

Reply:
done.